pkg/acme/util/util.go

```diff
@@ -21,42 +21,40 @@ import (
 	"math/big"
 	"net/http"
 	"time"
-
-	"github.com/jetstack/cert-manager/pkg/logs"
 )
 
-// RetryBackoff is the ACME client RetryBackoff that filters rate limit errors to our retry loop
-// inspired by acme/http.go
-func RetryBackoff(n int, r *http.Request, res *http.Response) time.Duration {
-	var jitter time.Duration
-	if x, err := rand.Int(rand.Reader, big.NewInt(1000)); err == nil {
-		// Set the minimum to 1ms to avoid a case where
-		// an invalid Retry-After value is parsed into 0 below,
-		// resulting in the 0 returned value which would unintentionally
-		// stop the retries.
-		jitter = (1 + time.Duration(x.Int64())) * time.Millisecond
-	}
-	if _, ok := res.Header["Retry-After"]; ok {
-		// if Retry-After is set we should
-		// error and let the cert-manager logic retry instead
-		return -1
-	}
-
-	// don't retry more than 10 times
-	if n > 10 {
-		return -1
+// RetryBackoff is the ACME client RetryBackoff that does not retry
+// all retries will be handled by cert-manager
```

Comment:
Sorry to pick nits, but the description here doesn't match the new code in this function.
I think it'd be worth updating the description to explain exactly how this function is meant to behave.
Explain, for example what circumstances we'd expect to receive that `badNonce` error, and why we *do* retry in that circumstance.
And why do we need to backoff  in the case of the `badNonce` error? The previous version of the function just retried every second (+jitter) for  up to 10 seconds.

I think we should copy your "comment for the reviewer" paragraph from the PR, into here, so that it's available for future maintainers of this code. 

Also, why does Pebble return `badNonce` errors by design?

Reply:
> Explain, for example what circumstances we'd expect to receive that badNonce error, and why we do retry in that circumstance.

ACME Spec says to ALWAYS expect it this is why pebble does it so often 

the way of calculating this comes from Go's code